### PR TITLE
refactor(behavior_velocity_template_module): prefix package and namespace with autoware_

### PR DESCRIPTION
**Pull request by esteve**
## Description

This PR adds the `autoware_` prefix to the package and puts headers in the `autoware` namespace.

Part of:
- https://github.com/autowarefoundation/autoware/issues/4569

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by esteve** [comment]
Back to draft because I'm going to rename folders and move headers to a separate folder (see https://github.com/autowarefoundation/autoware/issues/4569#issuecomment-2097690750)

**Comment by esteve** [comment]
@xmfcx @mitsudome-r I've renamed the base folder, this is now ready for review. Thanks!

File: planning/autoware_behavior_velocity_template_module/src/manager.hpp
Comment (xmfcx) [inline planning/autoware_behavior_velocity_template_module/src/manager.hpp:40]: When [behavior_velocity_planner_common/scene_module_interface.hpp/SceneModuleInterface](https://github.com/autowarefoundation/autoware.universe/blob/010965b9b1f8ba8056fdbe5e6a2aa8eb3e68e6a0/planning/behavior_velocity_planner_common/include/behavior_velocity_planner_common/scene_module_interface.hpp#L76) is moved under `autoware` namespace, this will need to be updated too right?

Just asking out of curiosity, right now it works and links correctly.

**Comment by xmfcx** [comment]
@danielsanchezaran please review it too :bow: , I will merge this by tomorrow.

File: planning/autoware_behavior_velocity_template_module/src/manager.hpp
Comment (esteve) [reply planning/autoware_behavior_velocity_template_module/src/manager.hpp:40]: Yes, I did it this way so that this package will fail when I refactor the other ones

**Comment by xmfcx** [comment]
Similar to https://github.com/autowarefoundation/autoware.universe/pull/6693#pullrequestreview-2048674718:
- https://github.com/autowarefoundation/autoware.universe/blob/bca574a9a37cd28616e146dca4b945a6cd8d8c4b/planning/.pages#L22C46-L22C79
- https://github.com/autowarefoundation/autoware.universe/blob/bca574a9a37cd28616e146dca4b945a6cd8d8c4b/.github/CODEOWNERS#L181

Needs to be updated.
cc. @maxime-clem

File: planning/autoware_behavior_velocity_template_module/src/scene.hpp
Comment (danielsanchezaran) [inline planning/autoware_behavior_velocity_template_module/src/scene.hpp:28]: Any reason for adding these lines? Anyways it LGTM. 
Please apply the changes @xmfcx requested too before merging.

**Review by danielsanchezaran — APPROVED** [review]

File: planning/autoware_behavior_velocity_template_module/src/scene.hpp
Comment (esteve) [reply planning/autoware_behavior_velocity_template_module/src/scene.hpp:28]: These two classes are in the `behavior_velocity_planner` namespace right now, whereas this package will be in `autoware::behavior_velocity_planner`, so I added these two `using` clauses so that the code in this PR can compile and also so that when the `behavior_velocity_planner` is refactored, the changes will be easier to track.

**Comment by esteve** [comment]
@xmfcx I've applied your feedback in https://github.com/autowarefoundation/autoware.universe/pull/6639/commits/f5ec966aebb2840fc35099ba2205315622ed3d64

**Review by xmfcx — CHANGES_REQUESTED** [review]
@esteve could you fix the pre-commit errors? https://results.pre-commit.ci/run/github/429651616/1715682704.OWIkFTVoQwqPf4XIzzdBaQ

> ```
> planning/static_centerline_generator/src/centerline_source/bag_ego_trajectory_based_centerline.cpp:27:  Add #include <string> for string  [build/include_what_you_use] [4]
> planning/static_centerline_generator/src/centerline_source/bag_ego_trajectory_based_centerline.cpp:44:  Add #include <memory> for make_shared<>  [build/include_what_you_use] [4]
> Done processing planning/static_centerline_generator/src/centerline_source/bag_ego_trajectory_based_centerline.cpp
> planning/static_centerline_generator/src/static_centerline_generator_node.cpp:601:  Add #include <algorithm> for min  [build/include_what_you_use] [4]
> Done processing planning/static_centerline_generator/src/static_centerline_generator_node.cpp
> planning/static_centerline_generator/src/utils.cpp:98:  Add #include <limits> for numeric_limits<>  [build/include_what_you_use] [4]
> Done processing planning/static_centerline_generator/src/utils.cpp
> Total errors found: 4
> planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp:171:  Add #include <algorithm> for max  [build/include_what_you_use] [4]
> Done processing planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp
> Total errors found: 1
> ```

**Comment by esteve** [comment]
@xmfcx I don't know what happened, but the changes for static_centerline_generator got mixed in. I've reverted them.

**Comment by xmfcx** [comment]
Could you rebase to latest main and fix the conflicts?

**Comment by esteve** [comment]
@xmfcx done, this branch had conflicts after https://github.com/autowarefoundation/autoware.universe/pull/6693 was merged

**Review by takayuki5168 — APPROVED** [review]

**Pull request merged**
